gbpservice/nfp/configurator/lib/schema_validator.py

```diff
@@ -1,12 +1,38 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
 from oslo_log import log
 import gbpservice.nfp.configurator.lib.schema as schema
 
 LOG = log.getLogger(__name__)
 
+""" Validates request data against standard resource schemas given in schema.py
+
+    Validation is focused on keys. It cross checks if resources in
+    request_data has all the keys given in the schema of that resource.
+"""
+
 
 class SchemaValidator():
 
     def decode(self, request_data):
+        """ Validate request data against resource schema.
+
+        :param: request_data
+
+        Returns: True - If schema validation is successful.
+                 False - If schema validation fails.
+
+        """
```

Comment:
We need to address the case where service_type is not passed in *aaS request. As per the current code, it will treat it as generic config but it will not really be.